openghg/plotting/_timeseries.py

```diff
@@ -95,28 +227,64 @@ def plot_timeseries(
             unit_conversion = float(data_units) / float(unit_value)
         else:
             unit_conversion = 1
+            # TODO: Not sure what is expected for unit_value here
+            unit_value = "1"
 
         y_data *= unit_conversion
 
-        fig.add_trace(go.Scatter(name=legend, x=x_data, y=y_data, mode="lines"))
+        unit_string = attributes_data["unit_print"][unit_value]
 
-    return fig
+        # Determine whether data is ascending or descending (positioning of legend)
+        ny = len(y_data)
+        data_left = y_data.isel(time=slice(int(ny * 0.9), -1)).mean()
```

Comment:
Just to check, are we able to assume the data will always be sorted? (I know `recombine_datasets` does this by default now but determining ascending/descending in this way does make that assumption).

As an alternative, ascending/descending could also be determined using [the xarray `DataArray.diff` method](https://docs.xarray.dev/en/stable/generated/xarray.DataArray.diff.html) which looks at the difference between each value in one go - you could also grab the mean of this or check every value is > or < 0 for instance.

Reply:
Added in this suggested change and a check for whether the overalll diff is positive or negative.